### PR TITLE
Mark is_rolemember functions PARALLEL RESTRICTED

### PR DESCRIPTION
**Pull request by ahmed-shameem**
### Description

JDBC tests which involve IS_ROLEMEMBER function fails with parallel query enabled. This is because internally this function calls function like get_physical_user_name which in turn checks the database name. As this involves database state, so we should not mark IS_ROLEMEMBER function parallel safe. As per [postgres documentation](https://www.postgresql.org/docs/current/parallel-safety.html#:~:text=miscellaneous%20backend%2Dlocal%20state%20that%20the%20system%20cannot%20synchronize%20across%20workers), a function must be marked as PARALLEL RESTRICTED if there is a possibility of miscellaneous backend-local state that the system cannot synchronize across workers. As get_physical_user_name function is accessing the database name, there is a possibility that the parallel workers may not be able to synchronize the database name among themselves.

### Issues Resolved

BABEL-4481, BABEL-4421

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/sql/sys_functions.sql [hunk 1]
```diff
@@ -3713,7 +3713,7 @@ CREATE OR REPLACE FUNCTION sys.is_rolemember_internal(
 	IN database_principal sys.SYSNAME
 )
 RETURNS INT AS 'babelfishpg_tsql', 'is_rolemember'
-LANGUAGE C STABLE PARALLEL SAFE;
+LANGUAGE C STABLE PARALLEL UNSAFE;
```
Comment (KushaalShroff) [inline contrib/babelfishpg_tsql/sql/sys_functions.sql]: We need to add these in upgrade files as well, right?

Comment (ahmed-shameem) [reply contrib/babelfishpg_tsql/sql/sys_functions.sql]: Updated in latest rev.

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql [hunk 2]
```diff
@@ -2994,6 +2994,31 @@ $BODY$
 LANGUAGE plpgsql IMMUTABLE PARALLEL UNSAFE;
 GRANT EXECUTE ON FUNCTION sys.FORMAT(IN anyelement, IN NVARCHAR, IN VARCHAR) TO PUBLIC;
 
+-- Role member functions
+CREATE OR REPLACE FUNCTION sys.is_rolemember_internal(
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql:4092]: how we reached to the conclusion that database name cannot be sync between workers?

Comment (kuntalghosh) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql:4092]: Yeah, worker will use the same role as the leader. So, we should be able to pushdown the rolename to parallel worker and make it parallel safe.

**Pull request closed without merging**